src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,85 @@
+import { Dispatch, SetStateAction } from 'react';
+import { Progress } from '../../types/Progress';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  itemsLeft: number,
+  setProgress: (type: Progress) => void;
+  isAnyCompleted: boolean,
+  todos: Todo[],
+  onClear: (todoId: number) => void,
+  setLoadingIds:Dispatch<SetStateAction<number[]>>
+}
+
+export const Footer:React.FC<Props> = ({
+  itemsLeft,
+  setProgress,
+  isAnyCompleted,
+  todos,
+  onClear,
+  setLoadingIds,
+}) => {
+  const ClearAllCompleted = async () => {
+    setLoadingIds((
+    ) => todos.filter(todo => todo.completed).map(todo => todo.id));
+
+    try {
+      await Promise.all(
+        todos
+          .filter(todo => todo.completed)
+          .map(async todo => {
+            await onClear(todo.id);
+          }),
```

Comment:
```suggestion
          .map(todo => onClear(todo.id)),
```
You await requests with `Promise.all`, so you can just create an array of promises and wait for the result like you do in `toggleAll` function